### ISSUE TITLE
Auto power on is not working for Cessna 208B Grand Caravan improvement mod

### ISSUE THREAD

**Issue by hawkeye-stan** [hawkeye-stan/msfs-popout-panel-manager#23]
Reported by Sling380 on flightsimulator.com:

When using C208 improvement mod operation. That aircraft requires the avionics switches on to power the G1000 screens and therefore the auto pop out panel doesn’t function. Powering up before manually starting the sequence works fine.

https://github.com/SheepCreativeSoftware/msfs2020-C208-Improvment-Mod

**Comment by hawkeye-stan** [comment]
After few hours of investigation, the mod has some bug in it as well as the original plane itself that causes pop out panel manager to fail.

1. Unlike the original plane, the mod version's panel does not pop out when power is off.  Also, it is sending the incorrect battery power on status which pop out manager depends on. On cold and dark, it is sending the SimConnect variable "ELECTRICAL MASTER BATTERY" as "true" indicating the plane is already powered on. But in the cockpit UI, the power switch is still off. Pop out manager needs the battery power signal as "off" to start the power on process. Also, my app can't run the power on/power off process blinding since if the user starts the flight on a runway, the power on will not do anything but then the power off will accidentally shuts off the plane.

2. Even if I get through the above issue, for some reason, this is the only G1000 plane in MSFS that is not accepting the 3 SimConnect SimEvents (MASTER_BATTERY_SET, AVIONICS_MASTER_1_ON, AVIONICS_MASTER_1_ON) the app sends to power on the battery and the avionics. The two avionics bus only accept K event of "ELECTRICAL_CIRCUIT_TOGGLE:26" and "ELECTRICAL_CIRCUIT_TOGGLE:27" which is custom to each plane. In the current iteration of pop out manager, even though I can sent one off events but I don't have a way yet to just send the above two K events to this plane only. If I blinding send these events to all planes, these planes will either randomly turn switches on and off at index 26 and 27 or totally breaks the plane javascript display.

In the meantime, I'll put it on my list and see if I can find a way to fix this plane.

**Comment by hawkeye-stan** [comment]
With more investigation yesterday night, I think I've found a solution. In my app's code, the existing SinConnect SimEvents to turn on Avionics Bus 2 for Cessna 172 (MFD) is also broke. It was definitely working pre SU9 or WU10 but I haven't flown this plane since these 2 updates.

What's broken in MSFS as of WU10:
1. ELECTRICAL MASTER BATTERY variable to detect cold and dark (broken for Cessna 208B grand caravan and mod)
2. MASTER_BATTERY_SET (working fine for all planes)
3. AVIONICS_MASTER_1_ON (working fine for all planes)
4. AVIONICS_MASTER_2_ON (broken for Cessna 208B grand caravan/mod + Cessna 172)

So after some trial and error, I found the SimEvents that work
1. To detect if a plane is cold and dark, I'm using a new variable "ATC ON PARKING SPOT". This variable changes to "true" a split second after clicking "Ready to Fly" button.  If the plane starts on runway, this variable is "false".
2. To turn on all avionics, I'm using a new event "TOGGLE_AVIONICS_MASTER" which now turns on all avionics for a plane (one or two avionics buses). This event wasn't working before but it is now working in WU10, weird!

And I've tested the new code in all G1000 planes (C172, C208, DA62, DA40NG, Bonanza G36, Pilatus PC-6, and my favorite plane Kodiak). I'll incorporate this change in the next update.
